### PR TITLE
`/fastq` API Additional filters

### PR DESCRIPTION
**Pull request by williamputraintan**
- Allow fastq list to be filtered based on `subjectId` , `workflow`, and `phenotype`.

Link PR: https://github.com/umccr/data-portal-client/pull/231

### CODE REVIEW

File: data_portal/models/fastqlistrow.py [hunk 1]
```diff
@@ -42,6 +42,27 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
...
             kwargs.pop('type')
 
+        type_ = kwargs.get('subject_id', None)
+        if type_:
+            q = self.reduce_multi_values_qor('subject_id', type_)
+            qs_meta = LabMetadata.objects.filter(q).values("library_id")
+            qs = qs.filter(rglb__in=qs_meta)
+            kwargs.pop('subject_id')
+
+        type_ = kwargs.get('phenotype', None)
```
Comment (victorskl) [inline data_portal/models/fastqlistrow.py:52]: ditto rename var name

File: data_portal/models/fastqlistrow.py [hunk 2]
```diff
@@ -42,6 +42,27 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
...
+        if type_:
+            q = self.reduce_multi_values_qor('phenotype', type_)
+            qs_meta = LabMetadata.objects.filter(q).values("library_id")
+            qs = qs.filter(rglb__in=qs_meta)
+            kwargs.pop('phenotype')
+
+        type_ = kwargs.get('workflow', None)
```
Comment (victorskl) [inline data_portal/models/fastqlistrow.py:59]: ditto rename var name

File: data_portal/models/tests/test_fastqlistrow.py [hunk 3]
```diff
@@ -56,3 +56,47 @@ def test_retrieve_by_project_owner(self):
         fqlr = results.get()
         logger.info(fqlr)
         self.assertEqual(fqlr.rglb, mock_lib_id)
+
+    def test_retrieve_by_subject_id(self):
```
Comment (victorskl) [inline data_portal/models/tests/test_fastqlistrow.py:60]: Nice! Thanks for writing test case.

**Review by victorskl — APPROVED** [review]
Minor comment to attend. Otherwise LGTM!

File: data_portal/models/fastqlistrow.py [hunk 4]
```diff
@@ -42,6 +42,27 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
             qs = qs.filter(rglb__in=qs_meta)
             kwargs.pop('type')
 
+        subject_id = kwargs.get('subject_id', None)
+        if subject_id:
+            q = self.reduce_multi_values_qor('subject_id', type_)
```
Comment (victorskl) [inline data_portal/models/fastqlistrow.py:47]: Still `type_` here.

```
q = self.reduce_multi_values_qor('subject_id', subject_id)
```

File: data_portal/models/fastqlistrow.py [hunk 5]
```diff
@@ -42,6 +42,27 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
...
+            qs_meta = LabMetadata.objects.filter(q).values("library_id")
+            qs = qs.filter(rglb__in=qs_meta)
+            kwargs.pop('subject_id')
+
+        phenotype = kwargs.get('phenotype', None)
+        if phenotype:
+            q = self.reduce_multi_values_qor('phenotype', type_)
```
Comment (victorskl) [inline data_portal/models/fastqlistrow.py:54]: ditto

File: data_portal/models/fastqlistrow.py [hunk 6]
```diff
@@ -42,6 +42,27 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
...
+            qs_meta = LabMetadata.objects.filter(q).values("library_id")
+            qs = qs.filter(rglb__in=qs_meta)
+            kwargs.pop('phenotype')
+
+        workflow = kwargs.get('workflow', None)
+        if workflow:
+            q = self.reduce_multi_values_qor('workflow', type_)
```
Comment (victorskl) [inline data_portal/models/fastqlistrow.py:61]: ditto

File: data_portal/models/fastqlistrow.py (hunk 4 above)
Comment (williamputraintan) [reply data_portal/models/fastqlistrow.py]: 🤦‍♂️

**Comment by victorskl** [comment]
Thanks for fixes. Good to go!

**Pull request merged**